### PR TITLE
Implement FCN8sAtOnce model with training code

### PR DESCRIPTION
**Pull request by wkentaro**

### CODE REVIEW

**Comment by volzkzg** [comment]
Error when run in pytorch 0.2.0

/home/bg455/.local/lib/python2.7/site-packages/torch/autograd/_functions/tensor.py:468: UserWarning: self and mask not broadcastable, but have the same number of elements.  Falling back to deprecated pointwise behavior.
  return tensor.masked_select(mask)
Segmentation fault (core dumped)

seems something wrong with `viz = fcn.utils.visualize_segmentation(lbl_pred=lp, lbl_true=lt, img=img, n_class=n_class)` in trainer.py, line 125

**Comment by volzkzg** [comment]
I looked into the file utils.py in FCN library, I think something wrong with cv2 library.

**Comment by wkentaro** [comment]
Hmm, there is no problem on my env..
